### PR TITLE
Ai road halo (optional feature) - add AI magenta shadow to any road that used to be an AI road

### PR DESCRIPTION
**Pull request by Bonkles**
One of the features we would like in-house is the ability to see, at a glance, what roads used to be AI Roads, even after they've been added to the map & saved. 

This PR does two things: 
1) Introduces styling awareness of the 'source' tag and adds a magenta shadow to any road that has 'source=maxar' or 'source=digitalglobe'. 
2) Hinges this feature on a new 'rapid_config.yaml' that lives in the /data directory, which has an explicit 'enable: true/false' directive to govern this feature. 

The feature must be enabled in the new rapid_config.yaml before building/deploying.
ai-road-glow
 

After speaking with @gaoxm we decided to leave the 'fail at build time' functionality as-is- if there is no rapid_config.yaml, the build will simply fail because the generated 'rapid_config.json' that we look for will be unavailable.

### CODE REVIEW

File: data/rapid_config.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+# Rapid configuration-  for customizing runtime behavior. 
+# The use case for this file is to enable/disable features without
+# needing to open/edit/rebuild the code. 
+rapid_config: 
+  # This setting styles any features in the map with a special shadow effect if 
+  # and only if they were originally presented to the user as an to be AI 
+  # features
+  style_previous_ai_features:
```
Comment (gaoxm) [inline data/rapid_config.yaml]: does this only apply to FB AI roads (with `source=digitalglobe` tag)? if so, let's call it `style_fb_ai_roads` to be more specific

File: modules/svg/lines.js [hunk 2]
```diff
@@ -110,6 +110,12 @@ export function svgLines(projection, context) {
             return scoreA - scoreB;
         }
 
+        var isAiRoad =  function(d){
```
Comment (gaoxm) [inline modules/svg/lines.js]: `getAIRoadStylingClass` may be a more precise name

File: modules/ui/raw_tag_editor.js [hunk 3]
```diff
@@ -282,8 +282,9 @@ export function uiRawTagEditor(context) {
 
         function isReadOnly(d) {
             // Read-only for source=digitalglobe on ML roads. TODO: switch to check on __fbid__
-            if (_entityID && _entityID.indexOf('w-') === 0 && d.key === 'source' && d.value === 'digitalglobe') {
-                return true;
+            if (_entityID && _entityID.indexOf('w-') === 0 && d.key === 'source' 
+                && (d.value === 'digitalglobe' || d.value === 'maxar')) {
+                    return true;
```
Comment (gaoxm) [inline modules/ui/raw_tag_editor.js]: I think it's OK to unindent `return true;` here; e.g. imaging the case we had more lines of codes within the block, we wouldn't want to further-indent the whole block in that case.

File: build_data.js [hunk 4]
```diff
@@ -673,6 +673,17 @@ function translationsToYAML(translations) {
 }
 
 
+function writeRapidConfig() {
+    var readRapidConfig = readFileProm('data/rapid_config.yaml', 'utf8');
```
Comment (gaoxm) [inline build_data.js]: will this break if `data/rapid_config.yaml` doesn't exist?

Comment (Bonkles) [reply build_data.js]: This line will not break at build time- the line that will break is actually the import of rapid_config.json inside `data/index.js`.

**Comment by Bonkles** [comment]
Okay, just pushed a new round of changes to address your requests @gaoxm!

**Comment by Bonkles** [comment]
As far as the tests go, I am working on fixing them in the 'grid' PR:(https://github.com/facebookincubator/RapiD/pull/26)

Once that one is merged travis should be back to normal.

File: data/rapid_config.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+
+# Rapid configuration-  for customizing runtime behavior. 
+# The use case for this file is to enable/disable features without
+# needing to open/edit/rebuild the code. 
+rapid_config: 
+  # This setting styles any features in the map with a special shadow effect if 
+  # and only if they were originally presented to the user as an to be AI
```
Comment (gaoxm) [inline data/rapid_config.yaml]: "presented to the user as AI features."

File: modules/svg/lines.js [hunk 6]
```diff
@@ -10,7 +10,7 @@ import { utilArrayFlatten, utilArrayGroupBy } from '../util';
 import { utilDetect } from '../util/detect';
 import _isEqual from 'lodash-es/isEqual';
 import _omit from 'lodash-es/omit';
-
+import { rapid_feature_config } from '../../data/'
```
Comment (gaoxm) [inline modules/svg/lines.js]: add an empty line below?

**Comment by Bonkles** [comment]
All righty- Travis builds / tests are now working for npm 8/10 so I'm going to squash and merge!

**Pull request merged**